### PR TITLE
Update `generate` for Vite-based projects

### PR DESCRIPTION
**Pull request by chawes13**
In support of https://github.com/LaunchPadLab/client-template/pull/477

**QA Steps**
- Clone this repo and run `git switch update-generate-for-vite`
- Run `yarn install & yarn link` in source repo (i.e., this one)
- Run `yarn link @launchpadlab/lp-subsection-generator` in target repo (e.g., onboarding-client#migrate-to-vite)
- Run `yarn` in target repo
- Run `yarn generate-subsection foos`
  - [x] Files generated for the foos subsection
- Manually import the FooRoutes in src/js/main/Routes.jsx
- Run `yarn start`
  - [x] `/foos` shows the index copy
  - [x] `/foos/1` shows the show copy 

**QA Cleanup**
- Run `yarn unlink` in source repo
- Run `yarn install --force` in target repo

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -6,4 +6,10 @@ Version 7 of the lp-subsection-generator introduced the following changes that m
 1. The lp-subsection-generator argument `subsection-name` is now required. Previously, this argument was optional and would default to `sub-section`.
 
 ## Migration to Version ^8.0.0
-Version 8 requires `@reduxjs/toolkit`, which is now standard across LPL's client templates.
\ No newline at end of file
+Version 8 requires `@reduxjs/toolkit`, which is now standard across LPL's client templates.
+
+## Migration to Version ^9.0.0
+Version 9 requires major upgrades for two peer dependencies: `@reduxjs/toolkit@^2.2.1` and `redux@^5.0.1`. Refer to the [Migrating to RTK 2.0 and Redux 5.0 documentation](https://redux-toolkit.js.org/usage/migrating-rtk-2) for more details.
```
Comment (chawes13) [inline MIGRATION.md:12]: I missed this on the last PR 😳

File: template/components/.keep
[the remark this answers is not in the document]
Comment (chawes13) [reply template/components/.keep:1]: The client template takes this approach with the `public` folder

**Review by bhennes2 — APPROVED** [review]
LGTM code-wise and passed my local QA using the branch of the `onboarding-client` repo. Way 2 go!

**Comment by chawes13** [comment]
@mwislek I just updated this for Ionic to support https://github.com/LaunchPadLab/ionic-client-template/pull/71. LMK if you have any feedback!

File: MIGRATION.md [hunk 2]
```diff
@@ -6,4 +6,10 @@ Version 7 of the lp-subsection-generator introduced the following changes that m
...
+
+## Migration to Version ^10.0.0
+Version 10 is built on the premise of supporting the Vite-based version of the client-template. The ionic-client-template has not yet been migrated to Vite, so the files that are output are still based on the existing Webpack-based setup. If you plan on using this with a non-ionic project, ensure that you are using at least `v14` of https://github.com/LaunchPadLab/client-template.
```
Comment (mwislek) [inline MIGRATION.md:15]: Guess this can be updated!

**Review by mwislek — APPROVED** [review]
Other than minor migration comment approved! I tried this both for the client and ionic templates and it worked flawlessly. Nicely done!

**Pull request merged**